cider/tests/nextgen/jitlib/JITLibTest.cpp

```diff
@@ -143,6 +143,169 @@ TEST_F(JITLibTests, LogicalOpTest) {
       false, true, true, [](JITValue& a, JITValue& b) { return !a; });
 }
 
+TEST_F(JITLibTests, BasicIFControlFlowWithoutElseTest) {
+  LLVMJITModule module("TestModule");
+  JITFunctionPointer function = module.createJITFunction(JITFunctionDescriptor{
+      .function_name = "test_func",
+      .ret_type = JITFunctionParam{.type = JITTypeTag::INT32},
+      .params_type = {JITFunctionParam{.name = "x", .type = JITTypeTag::INT32},
+                      JITFunctionParam{.name = "condition", .type = JITTypeTag::BOOL}}});
+  {
+    JITValuePointer ret = function->createVariable("ret", JITTypeTag::INT32);
+    ret = function->getArgument(0);
+    auto if_builder = function->getIfBuilder();
+    if_builder->build(
+        [&]() {
+          auto condition = function->getArgument(1);
+          return condition;
+        },
+        [&]() { ret = ret + 1; });
+    function->createReturn(ret);
+  }
+  function->finish();
+  module.finish();
+
+  auto func_ptr = function->getFunctionPointer<int32_t, int32_t, bool>();
+  EXPECT_EQ(func_ptr(123, false), 123);
+  EXPECT_EQ(func_ptr(123, true), 124);
+}
+
+TEST_F(JITLibTests, BasicIFControlFlowWithElseTest) {
+  LLVMJITModule module("TestModule");
+  JITFunctionPointer function = module.createJITFunction(JITFunctionDescriptor{
```

Comment:
add pseudo code in the comment may be more clear?
```cpp
// func(int32 x, bool condition) {
//     if (condition) {
//         x = x + 1;
//     } else {
//         x = x + 10;
//     }
//     return x;
// }
```

Reply:
Seems the code structure is more clear after using pattern advised by chaojun.